userland/src/sync_parameters.cpp

```diff
@@ -0,0 +1,96 @@
+// Copyright 2015 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <algorithm>
+#include <chrono>
+#include <cmath>
+#include <iostream>
+#include <string>
+#include <thread>
+
+#include <rclcpp/rclcpp.hpp>
+
+#include <simple_msgs/AllBuiltinTypes.h>
+#include <simple_msgs/AllDynamicArrayTypes.h>
+#include <simple_msgs/AllPrimitiveTypes.h>
+#include <simple_msgs/AllStaticArrayTypes.h>
+#include <simple_msgs/Nested.h>
+#include <simple_msgs/String.h>
+#include <simple_msgs/Uint32.h>
+
+#include <userland_msgs/AddTwoInts.h>
+
+int main(int argc, char ** argv)
+{
+  rclcpp::init(argc, argv);
+
+  auto node = rclcpp::Node::make_shared("parameters");
+
+  auto sync_parameters_client = std::make_shared<rclcpp::parameter::SyncParametersClient>(node);
+
+  auto parameters = {
+    rclcpp::parameter::ParameterVariant("foo", int64_t(2)),
+    rclcpp::parameter::ParameterVariant("bar", std::string("hello")),
+    rclcpp::parameter::ParameterVariant("baz", double(1.45)),
+    rclcpp::parameter::ParameterVariant("foobar", true),
+  };
+
+  auto results1 = sync_parameters_client->set_parameters(parameters);
+  for (auto v : results1) {
+    if (!v.successful) {
+      std::cerr << v.reason << std::endl;
+    }
+  }
+
+  auto results2 = sync_parameters_client->get_parameters({{"foo", "baz"}});
```

Comment:
Yeah, I guess if `get_parameters` could take either a vector of strings or a string then `get_parameters({"foo"})` could be considered ambiguous. I think a similar scenario was the issue on Windows, but clang and gcc correctly (imo) say `get_parameters({"foo"})` matches the string version and `get_parameters({{"foo"}})` would be needed to match the vector of strings.

If we wanted to be explicit then we would want to do `get_parameters({{"foo"}, {"bar"}})`.

But since `get_parameters` doesn't yet take just a string, I think your latest commit is perfect.